src/main/java/calculationEngine/environment/CeItem.java

```diff
@@ -1,14 +1,97 @@
 package calculationEngine.environment;
 
 public class CeItem {
-    private final int bonus; //Just a placeholder until we have a correct structure for items.
 
-    public CeItem(int bonus) {
-        this.bonus = bonus;
+    private String name;
+    private String spriteName;
+    private boolean equipped;
+    private boolean unique;
+    private CeItemTypes type;
+    private int uses;
+    private CeItemBonusStats itemBonusStats;
+    private String description;
+
+    public CeItem() {
+    }
+
+    public void setItemFromSaveGame(String name, String spriteName, boolean equipped, boolean unique, String type, int uses, int[] itemBonusStats, String description) {
+        this.name = name;
+        this.spriteName = spriteName;
+        this.equipped = equipped;
+        this.unique = unique;
+        this.type = CeItemTypes.valueOf(type);
+        this.uses = uses;
+        this.itemBonusStats = new CeItemBonusStats(itemBonusStats[0], itemBonusStats[1], itemBonusStats[2], itemBonusStats[3], itemBonusStats[4], itemBonusStats[5], itemBonusStats[6]);
+        this.description = description;
+    }
+
+    public void setNewLootedItem(String name, String spriteName, boolean unique, String type, int uses, int[] itemBonusStats, String description) {
+        this.name = name;
```

Comment:
why aren't you using constructors for this?

Reply:
Because inside our CE we have a literal Constructor hell.. every class needs several constructors for different use case and every time we have to use them, you have to think about the specific use case for every single constructor in order to understand what it's supposed to do.. also Code Completion gets almost unusable, because of the masses of different signatures a single class has (in some cases we have 7 Constructors for one Class). In order to tackle this problem and keep our code somewhat readable, i've used this approach.

Reply:
Ok, understandable..

Reply:
Ok, Understandable